website/static/js/profile.js

```diff
@@ -563,13 +572,86 @@ var SocialViewModel = function(urls, modes) {
         return false;
     });
 
+    self.addWebsiteInput = function() {
+        this.profileWebsites.push(ko.observable().extend({
+            ensureHttp: true
+        }));
+    };
+    
+    self.removeWebsite = function(profileWebsite) {
+        var profileWebsites = ko.toJS(self.profileWebsites()),
+            idx = profileWebsites.indexOf(profileWebsite);
+            bootbox.confirm({
+                title: 'Remove website?',
+                message: 'Are you sure you want to remove this website from your profile?',
+                callback: function(confirmed) {
+                    if (confirmed) {
+                        self.profileWebsites.splice(idx, 1);
+                        if (self.profileWebsites().length === 0) {
+                            self.addWebsiteInput();
+                        }
+                        self.submit();
+                        self.changeMessage(
+                            'Website Removed',
+                            'text-danger',
+                            5000
+                        );
+                    }
+                },
+                buttons:{
+                    confirm:{
+                        label:'Remove',
+                        className:'btn-danger'
+                    }
+                }
+            });
+    };
+
+    if (self.profileWebsites().length === 0) {
```

Comment:
What's the purpose of this conditional? It seems out of place -- please document it's purpose or remove in no longer needed